src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1515 @@
+
+        eligible_population = df.is_alive & (df.sex == 'F') & (df.age_years > 30) & (df.age_years < 50) & \
+                              ~df.ce_current_cc_diagnosed
+
+        df.loc[eligible_population, 'ce_selected_for_via_this_month'] = (
+            np.random.random_sample(size=len(df[eligible_population])) < p['prob_via_screen']
+        )
+
+        df.loc[eligible_population, 'ce_selected_for_xpert_this_month'] = (
+            np.random.random_sample(size=len(df[eligible_population])) < p['prob_xpert_screen']
+        )
+
+        self.sim.modules['SymptomManager'].change_symptom(
+            person_id=df.loc[df['ce_selected_for_via_this_month']].index,
+            symptom_string='chosen_via_screening_for_cin_cervical_cancer',
+            add_or_remove='+',
+            disease_module=self.module
+        )
+
+        self.sim.modules['SymptomManager'].change_symptom(
+            person_id=df.loc[df['ce_selected_for_xpert_this_month']].index,
+            symptom_string='chosen_xpert_screening_for_hpv_cervical_cancer',
+            add_or_remove='+',
+            disease_module=self.module
+        )
+
+
+
+
+    # -------------------- UPDATING OF SYMPTOM OF vaginal bleeding OVER TIME --------------------------------
+        # Each time this event is called (every month) individuals with cervical cancer may develop the symptom of
+        # vaginal bleeding.  Once the symptom is developed it never resolves naturally. It may trigger
+        # health-care-seeking behaviour.
+        onset_vaginal_bleeding = self.module.lm_onset_vaginal_bleeding.predict(
+            df.loc[
+                np.bitwise_and(df.is_alive, df.ce_stage_at_diagnosis == 'none')
+            ],
+            rng
+        )
+
+        self.sim.modules['SymptomManager'].change_symptom(
+            person_id=onset_vaginal_bleeding[onset_vaginal_bleeding].index.tolist(),
+            symptom_string='vaginal_bleeding',
+            add_or_remove='+',
+            disease_module=self.module
+        )
+
+
+# vaccinating 9 year old girls - this only uncommented for testing - vaccination is controlled by epi
+#       age9_f_idx = df.index[(df.is_alive) & (df.age_exact_years > 9) & (df.age_exact_years < 90) & (df.sex == 'F')]
+#       df.loc[age9_f_idx, 'va_hpv'] = 1
+
+        # -------------------- DEATH FROM cervical CANCER ---------------------------------------
+        # There is a risk of death for those in stage4 only. Death is assumed to go instantly.
+        stage4_idx = df.index[df.is_alive & (df.ce_hpv_cc_status == "stage4")]
+        selected_to_die = stage4_idx[
+            rng.random_sample(size=len(stage4_idx)) < self.module.parameters['r_death_cervical_cancer']]
+
+        for person_id in selected_to_die:
+            self.sim.schedule_event(
+                InstantaneousDeath(self.module, person_id, "CervicalCancer"), self.sim.date
+            )
+            df.loc[selected_to_die, 'ce_date_death'] = self.sim.date
```

Comment:
is date of death not recorded already in demography?

Reply:
You don't need to store the date of death yourself, and it's stored in the demography log. That said, there's no harm to it, other than memory wastage.